staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go

```diff
@@ -21,9 +21,23 @@ import (
 	"fmt"
 )
 
-var ErrBadJSONDoc = errors.New("Invalid JSON document")
-var ErrNoListOfLists = errors.New("Lists of lists are not supported")
-var ErrBadPatchFormatForPrimitiveList = errors.New("Invalid patch format of primitive list")
+var (
+	ErrBadJSONDoc                     = errors.New("Invalid JSON document")
+	ErrNoListOfLists                  = errors.New("Lists of lists are not supported")
+	ErrBadPatchFormatForPrimitiveList = errors.New("Invalid patch format of primitive list")
+)
+
+func ErrNoMergeKey(m map[string]interface{}, k string) error {
+	return fmt.Errorf("map: %v does not contain declared merge key: %s", m, k)
+}
+
+func ErrBadArgType(expected, actual string) error {
+	return fmt.Errorf("expected a %s, but received a %s", expected, actual)
```

Comment:
`%T` gives a type that is too concrete for us. Return a very concrete type may confuse the users sometimes. 
```
x := []byte("foo")
fmt.Printf("%T\n", x)
fmt.Printf(reflect.TypeOf(x).Kind().String())

[]uint8
slice
```